### PR TITLE
Replace Glide with Dep

### LINKED ISSUE #483: Switch from Glide to Dep
Is there any plan to switch Go dependency management tool from Glide to Dep? 

Dep announced that it is safe for production use. see https://github.com/golang/dep#current-status

And the Glide project is suggesting users consider trying to migrate from Glide to dep. see https://github.com/Masterminds/glide#golang-dep
> Glide will continue to be supported for some time but is considered to be in a state of support rather than active feature development.

### PR DESCRIPTION
**Pull request by frankgreco**
- [x] Generate correct `Gopkg.toml` and `Gopkg.lock`.
  - [x] Update `glide` references in `Makefile` and replace with `dep` equivalents.
  - [x] Update documentation.

Fixes #483

### CODE REVIEW

**Comment by vprithvi** [comment]
Looks good to me so far, the DCO check fails because it's looking for sign offs in all commits. 

Also, was the toml file generated from `glide.yaml`, if so how?

**Comment by frankgreco** [comment]
@vprithvi I forgot the `-s` flag in my `git commit` in my first commit. How would I modify it after the fact? Also, I manually generated the `Gopkg.toml` from the `Glide.yaml` file (very straight forward). The `Gopkg.lock` file was generated using `dep ensure`

**Comment by frankgreco** [comment]
@vprithvi in an attempt to signoff on my first commit, I went `HEAD~<one too many>` and so I incorrectly modified the signing of @black-adder commit. Any idea on how to fix this?

File: Makefile [hunk 1]
```diff
@@ -83,23 +84,21 @@ fmt:
...
 	@[ ! -s "$(FMT_LOG)" ] || (echo "Go fmt or license check failures, run 'make fmt'" | cat - $(FMT_LOG) && false)
 
-.PHONY: install-glide
-install-glide:
-	# all we want is: glide --version || go get github.com/Masterminds/glide
-	# but have to pin to 0.12.3 due to https://github.com/Masterminds/glide/issues/745
-	@which glide > /dev/null || (mkdir -p $(GOPATH)/src/github.com/Masterminds && cd $(GOPATH)/src/github.com/Masterminds && git clone https://github.com/Masterminds/glide.git && cd glide && git checkout v0.12.3 && go install)
+.PHONY: install-dep
+install-dep:
+	@which dep > /dev/null || (curl -L -s -o dep https://github.com/golang/dep/releases/download/$(DEP_VER)/dep-linux-amd64 && chmod +x dep && sudo mv dep /usr/local/bin/)
```
Comment (glibsm) [inline Makefile]: Should this download link take into account host platform?

Comment (frankgreco) [reply Makefile]: yea good point

**Comment by vprithvi** [comment]
> @vprithvi in an attempt to signoff on my first commit, I went HEAD~<one too many> and so I incorrectly modified the signing of @black-adder commit. Any idea on how to fix this?

You could create a new branch from master, cherry pick the range of your commits into that new branch, and force push that branch to your old remote. 

I'm not sure, but you might also be able to do an interactive rebase off master, and remove the line corresponding to the modified commit that wasn't yours.

**Comment by jpkrohling** [comment]
Is this still relevant? Do we want still want to get rid of Glide, and if so, is `dep` still the right move? Or can this be closed?

**Comment by isaachier** [comment]
I doubt we want to get rid of glide. Unless someone knows better, this looks done to me.

**Comment by objectiser** [comment]
As glide project is in maintenance mode, and recommending projects move to dep, I think it is only a case of when this change will be made. So wouldn't it be better to leave this issue open?

**Comment by isaachier** [comment]
In theory, sure. In practice, Uber still uses glide for 99% of internal projects so we'd probably keep this around until that number drops significantly.

**Comment by yurishkuro** [comment]
Dep is now officially supported at Uber internally, we should upgrade the internal build to dep and then we can kill glide here as well. But then Go officially moving to vgo/modules, I haven't even looked at those.

**Comment by jpkrohling** [comment]
> But then Go officially moving to vgo/modules, I haven't even looked at those.

That's what I had in mind when I asked if moving to `dep` was the right move :)

**Comment by isaachier** [comment]
Right it's really sad dep didn't pan out. I don't mind if we kill glide as long as dep is fully supported internally and works just as well as glide.

**Comment by isaachier** [comment]
What's up with this?

**Comment by justinclift** [comment]
Should this be closed, now that #1240 is merged?

**Pull request closed without merging**